cmd/podman/artifact/rm.go

```diff
@@ -11,40 +11,62 @@ import (
 
 var (
 	rmCmd = &cobra.Command{
-		Use:               "rm ARTIFACT",
-		Short:             "Remove an OCI artifact",
-		Long:              "Remove an OCI from local storage",
-		RunE:              rm,
-		Aliases:           []string{"remove"},
-		Args:              cobra.ExactArgs(1),
+		Use:     "rm [options] ARTIFACT",
+		Short:   "Remove an OCI artifact",
+		Long:    "Remove an OCI from local storage",
+		RunE:    rm,
+		Aliases: []string{"remove"},
+		Args: func(cmd *cobra.Command, args []string) error { //nolint: gocritic
+			return checkAllAndArgs(cmd, args)
+		},
 		ValidArgsFunction: common.AutocompleteArtifacts,
-		Example:           `podman artifact rm quay.io/myimage/myartifact:latest`,
-		Annotations:       map[string]string{registry.EngineMode: registry.ABIMode},
+		Example: `podman artifact rm quay.io/myimage/myartifact:latest
+podman artifact rm -a`,
+		Annotations: map[string]string{registry.EngineMode: registry.ABIMode},
 	}
-	// The lint avoid here is because someday soon we will need flags for
-	// this command
-	rmFlag = rmFlagType{} //nolint:unused
+
+	rmOptions = entities.ArtifactRemoveOptions{}
 )
 
-// TODO at some point force will be a required option; but this cannot be
-// until we have artifacts being consumed by other parts of libpod like
-// volumes
-type rmFlagType struct { //nolint:unused
-	force bool
+func rmFlags(cmd *cobra.Command) {
+	flags := cmd.Flags()
+	flags.BoolVarP(&rmOptions.All, "all", "a", false, "Remove all artifacts")
 }
-
 func init() {
 	registry.Commands = append(registry.Commands, registry.CliCommand{
 		Command: rmCmd,
 		Parent:  artifactCmd,
 	})
+	rmFlags(rmCmd)
 }
 
 func rm(cmd *cobra.Command, args []string) error {
-	artifactRemoveReport, err := registry.ImageEngine().ArtifactRm(registry.Context(), args[0], entities.ArtifactRemoveOptions{})
+	var nameOrID string
+	if len(args) > 0 {
+		nameOrID = args[0]
```

Comment:
What about multiple arguments being passed? Should we loop over args?

Reply:
checkAllAndArgs() checks that prior

Reply:
I don't see that check. It just verifies it's more than 0 - there's no check that it is exactly 1. As written, it seems arguments past 1 are discarded silently.

Reply:
@Luap99 made the same basic comment and I have it fixed locally